### PR TITLE
[pr2eus_tutorials] add pr2 tabletop demo test and sample

### PR DESCRIPTION
**Pull request by furushchev**
need https://github.com/jsk-ros-pkg/jsk_roseus/pull/436

### CODE REVIEW

**Comment by furushchev** [comment]
update to use jsk_pcl version of tabletop_object_detection
This PR needs https://github.com/jsk-ros-pkg/jsk_recognition/pull/1585

**Comment by k-okada** [comment]
do we need to release jsk_recognition?

--
◉ Kei Okada

**Comment by furushchev** [comment]
@k-okada Yes!

Yuki Furuta

14 апр. 2016 г., 15:09 +0900, Kei Okada<notifications@github.com>, писал:

**Comment by k-okada** [comment]
done https://github.com/ros/rosdistro/pull/11080

--
◉ Kei Okada

**Comment by furushchev** [comment]
needs https://github.com/jsk-ros-pkg/jsk_robot/pull/591

**Comment by k-okada** [comment]
ok released
https://github.com/jsk-ros-pkg/jsk_robot/commit/00ee307330184e5ee2175bdcd5582adb3a89f154

--
◉ Kei Okada

**Comment by furushchev** [comment]
Build failed. restarting


```bash

executing command [sudo -H apt-get install -y -qq ros-indigo-jsk-pr2-startup]

E: Unable to locate package ros-indigo-jsk-pr2-startup

ERROR: the following rosdeps failed to install

  apt: command [sudo -H apt-get install -y -qq ros-indigo-jsk-pr2-startup] failed
```

**Comment by k-okada** [comment]
On Fri, May 6, 2016 at 11:24 PM, Furushchev <notifications@github.com>
wrote:

>   apt: command [sudo -H apt-get install -y -qq ros-indigo-jsk-pr2-startup] failed
>
>
re-run travis

--
◉ Kei Okada

**Comment by furushchev** [comment]
restarted 3 times. still jsk_pr2_startup is unavailable.
something broken?
http://repositories.ros.org/status_page/ros_indigo_default.html?q=jsk_pr2_startup

```
executing command [sudo -H apt-get install -y -qq ros-indigo-jsk-pr2-startup]

E: Unable to locate package ros-indigo-jsk-pr2-startup

ERROR: the following rosdeps failed to install

  apt: command [sudo -H apt-get install -y -qq ros-indigo-jsk-pr2-startup] failed
```

**Comment by furushchev** [comment]
23:01:58 CMake Error at cmake_install.cmake:86 (FILE):
23:01:58   file INSTALL cannot make directory
23:01:58   "/tmp/binarydeb/ros-indigo-jsk-pr2-startup-1.0.5/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/./debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/debian/ros-indigo-jsk-pr2-startup/opt/ros/indigo/share/jsk_pr2_startup/obj-x86_64-linux-gnu/CMakeFiles/geometry_msgs_generate_messages_py.dir":
23:01:58   File name too long
23:01:58 
23:01:58 
23:01:58 make[2]: *** [install] Error 1
23:01:58 dh_auto_install: make -j1 install DESTDIR=/tmp/binarydeb/ros-indigo-jsk-pr2-startup-1.0.5/debian/ros-indigo-jsk-pr2-startup AM_UPDATE_INFO_DIR=no returned exit code 2
23:01:58 make[1]: *** [override_dh_auto_install] Error 2
23:01:58 make: *** [binary] Error 2
23:01:58 dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2
23:01:58 E: Building failed
23:01:58 Traceback (most recent call last):
23:01:58   File "/tmp/ros_buildfarm/ros_buildfarm/binarydeb_job.py", line 120, in build_binarydeb
23:01:58     subprocess.check_call(cmd, cwd=source_dir)
23:01:58   File "/usr/lib/python3.4/subprocess.py", line 561, in check_call
23:01:58     raise CalledProcessError(retcode, cmd)
23:01:58 subprocess.CalledProcessError: Command '['apt-src', 'build', 'ros-indigo-jsk-pr2-startup']' returned non-zero exit status 1
23:01:58 
23:01:58 --------------------------------------------------------------------------------------------------
23:01:58 `apt-src build ros-indigo-jsk-pr2-startup` failed.
23:01:58 This is usually because of an error building the package.
23:01:58 The traceback from this failure (just above) is printed for completeness, but you can ignore it.
23:01:58 You should look above `E: Building failed` in the build log for the actual cause of the failure.
23:01:58 --------------------------------------------------------------------------------------------------

**Comment by furushchev** [comment]
refer: https://github.com/jsk-ros-pkg/jsk_robot/pull/618

**Comment by furushchev** [comment]
found error

```
fatal: Not a git repository: /var/lib/jenkins/jobs/trusty-travis-jsk-ros-pkg-jsk_pr2eus-indigo-deb-true/workspace/jenkins-trusty-travis-jsk-ros-pkg-jsk_pr2eus-indigo-deb-true-433/jsk-ros-pkg/jsk_pr2eus/.git/modules/.travis
```

restarted.

**Comment by furushchev** [comment]
oh, jsk-ros-pkg/jsk_robot#618 is not released yet.

**Comment by knorth55** [comment]
What is the status of this PR?

File: pr2eus_tutorials/README.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+
+```bash
+# on local machine
+rosrun pr2eus_tutorials pr2_tabletop_sim.sh # PLEASE NOTE NOT `roslaunch` BUT `rosrun`!
+```
+
+In sample code, pr2 brings up detected object.
+
+```bash
+rosrun pr2eus_tutorials pr2-tabletop-sample.l
```
Comment (knorth55) [inline pr2eus_tutorials/README.md]: `pr2-tabletop-object-grasp.l`

**Comment by k-okada** [comment]
why test fails?

**Comment by furushchev** [comment]
@k-okada I modified the test code and tested on my laptop and found that a robot does not move when euslisp code is executed as `test` node and I have no idea what is the actual problem...
So I just did not commit the update for test codes and disabled. (Sorry for no description about that)

The weird thing is if I run normally with roslaunch, it works.

- `roslaunch` test code (it works on my laptop)

```bash
# terminal 1
roscd pr2eus_tutorials/test
roslaunch test_pr2_tabletop_sim.test
```

```bash
# terminal 2
roscd pr2eus_tutorials/test
roseus test-pr2-tabletop.l
```

- `rostest` (It fails...)

```bash
roscd pr2eus_tutorials/test
rostest test_pr2_tabletop_sim.test
```

Any idea?

**Comment by furushchev** [comment]
I just now committed my update for test code.
If the problem is not solved, I'll just revert the latest commit to disable test.

**Comment by knorth55** [comment]
closed via #387

**Pull request closed without merging**